ReactWindows/ReactNative.Shared/UIManager/UIImplementation.cs

```diff
@@ -658,7 +659,9 @@ public void OnShutdown()
 
         private void UpdateViewHierarchy()
         {
-            foreach (var tag in _shadowNodeRegistry.RootNodeTags)
+            var rootNodeTags = new List<int>(_shadowNodeRegistry.RootNodeTags.ToList());
```

Comment:
The InvalidOperaitonException likely happens because the ShadowNoteRegistry is being updated while the UpdateViewHierachy is iterating over the root node tag list.  One possibility is that RemoveRootNode is being called when iteration is in progress. More likely, it is the AddRootNode that is being called.  However, it's not clear whether it's possible for AddRootNode to be called twice for the same node since there is no check that the tag is already registered.

To avoid the synchronization issue around access to the RootTags dictionary I propose making a copy of the Keys collection and iterating over the copy instead of directly on the dictionary.  This should prevent the InvalidOperaitonException.